### PR TITLE
fix(agw): Converted gtpv1 task files to cpp

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
fix(agw): Converted gtpv1-u task files to cpp [Extension to Github Issue: #13096]

## Summary
Migrated all the files related to gtpv1-u task in the folders `magma/lte/gateway/c/core/oai/tasks/gtpv1-u/` and `magma/lte/gateway/c/core/oai/include/` to `cpp/hpp` and fixed all the compilation errors and warnings.

Also followed the guidelines for order of headers from https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

## Test plan
Verified compilation with following:
- make clean; make build_oai;
- make clean; bazel build //lte/gateway/c/core:agw_of
~~make clean; bazel build --config=production //lte/gateway/c/core:agw_of~~
- make clean; bazel test //lte/gateway/c/core/...
~~make clean; bazel test //lte/gateway/c/core/oai/test/amf:amf_stateless_test~~
- make clean; make FEATURES=mme_oai build_oai
~~make clean; export FEATURES=mme_oai; make build_oai~~

These compilations are no more valid on latest master after removal of target test_oai from Make file:
```
make clean; make test_oai;
make clean; make test_oai BUILD_TYPE=Debug OAI_TESTS=test_amf_app
```

Verified with sanity and in CI

Signed-off-by: VinashakAnkitAman <ankit.aman@radisys.com>

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp [hunk 1]
```diff
@@ -609,9 +610,9 @@ static void insert_sgw_cp_and_up_teid_to_directoryd(sgw_state_t* sgw_state,
              sizeof(teidString) - strlen(teidString), "%u%c", teidlist[idx],
              separator);
   }
-  char* ptr = "sgw_c_teid";
+  char* ptr = const_cast<char*>("sgw_c_teid");
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp]: Why not using a const char*?
```cpp
const char* ptr = "sgw_c_teid";
```

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp [hunk 2]
```diff
@@ -609,9 +610,9 @@ static void insert_sgw_cp_and_up_teid_to_directoryd(sgw_state_t* sgw_state,
...
   if (teid_type == UP_TEID) {
-    ptr = "sgw_u_teid";
+    ptr = const_cast<char*>("sgw_u_teid");
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp]: Change is not necessary when using const char*.

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_openflow.cpp [hunk 3]
```diff
@@ -180,21 +188,21 @@ static uint32_t create_gtp_port(struct in_addr enb_addr,
...
-    gtp_csum = "false";
+    gtp_csum = const_cast<char*>("false");
   }
 
   if (is_pgw && spgw_config.sgw_config.agw_l3_tunnel) {
-    l3_tunnel = "true";
+    l3_tunnel = const_cast<char*>("true");
   } else {
-    l3_tunnel = "false";
+    l3_tunnel = const_cast<char*>("false");
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_openflow.cpp:205]: Changes are unnecessary when replacing
https://github.com/magma/magma/blob/01ad2d866bfa93926e5eb4ad93418348faf7917e/lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_openflow.cpp#L184-L186
with
```cpp
const char* gtp_echo;
const char* gtp_csum;
const char* l3_tunnel;
```

File: lte/gateway/c/core/oai/lib/openflow/controller/GTPApplication.cpp [hunk 4]
```diff
@@ -21,7 +21,7 @@
 
 #include "lte/gateway/c/core/oai/lib/openflow/controller/GTPApplication.hpp"
```
Comment (ajahl) [inline lte/gateway/c/core/oai/lib/openflow/controller/GTPApplication.cpp]: Please move it to the top of the includes.

File: lte/gateway/c/core/oai/lib/openflow/controller/GTPApplication.cpp [hunk 5]
```diff
@@ -21,7 +21,7 @@
...
 #include "lte/gateway/c/core/oai/lib/openflow/controller/IMSIEncoder.hpp"
-#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u.h"
+#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u.hpp"
```
Comment (ajahl) [inline lte/gateway/c/core/oai/lib/openflow/controller/GTPApplication.cpp]: Can be removed. The include is already made in `GTPApplication.hpp`.

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u_task.cpp [hunk 6]
```diff
@@ -26,24 +26,24 @@
...
-#endif
-#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u_sgw_defs.h"
 #ifdef __cplusplus
 }
 #endif
 
+#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_upf.hpp"
+#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u.hpp"
+#include "lte/gateway/c/core/oai/tasks/sgw/pgw_ue_ip_address_alloc.hpp"
+#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u_sgw_defs.hpp"
```
Comment (pruthvihebbani) [inline lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u_task.cpp:46]: Move gtpv1u_sgw_defs.hpp before pgw_ue_ip_address_alloc.hpp

File: lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp [hunk 7]
```diff
@@ -70,6 +69,7 @@ extern "C" {
 #include "lte/gateway/c/core/oai/tasks/sgw/pgw_procedures.hpp"
 #include "lte/gateway/c/core/oai/tasks/sgw/pgw_ue_ip_address_alloc.hpp"
 #include "lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.hpp"
+#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u.hpp"
```
Comment (pruthvihebbani) [inline lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp]: Move sgw_handlers.hpp to top

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp [hunk 8]
```diff
@@ -36,6 +35,7 @@ extern "C" {
 #include "lte/gateway/c/core/oai/tasks/sgw/pgw_procedures.hpp"
 #include "lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.hpp"
 #include "lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_s11_handlers.hpp"
+#include "lte/gateway/c/core/oai/tasks/gtpv1-u/gtpv1u.hpp"
```
Comment (pruthvihebbani) [inline lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp]: Move sgw_s8_s11_handlers.hpp to top

**Review by pruthvihebbani — CHANGES_REQUESTED** [review]

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp (hunk 8 above)
Comment (VinashakAnkitAman) [reply lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp]: @pruthvihebbani This is not own header. Isn't it? The own header should be "sgw_s8_handlers.hpp". Do I need to move it to the top?

**Review by LKreutzer — APPROVED** [review]
Bazel changes lgtm!

**Review by pruthvihebbani — APPROVED** [review]
LGTM

**Pull request merged**